### PR TITLE
Betting session handling.

### PR DESCRIPTION
**Pull request by pdex**
Put SessionContext objects back into the session pool. This will allow
us to re-use a single Database object.

### CODE REVIEW

File: src/main/java/com/google/spannerclient/Database.java [hunk 1]
```diff
@@ -92,13 +97,20 @@
         new FutureCallback<List<Session>>() {
           @Override
           public void onSuccess(List<Session> sessionList) {
+            // TODO(pdex): sessionList.size will always be equal to poolSize
+            //             remove this check and replace with a session healthcheck
```
Comment (xjdr) [inline src/main/java/com/google/spannerclient/Database.java]: This isn't true. Sessions can be GC'd when they are inactive for 60 min. We need to be able to add new sessions to maintain the pool size if it gets GC'd or failed.

File: src/main/java/com/google/spannerclient/GrpcClient.java [hunk 2]
```diff
@@ -145,6 +148,7 @@ void executeStreamingSql(
     Preconditions.checkNotNull(ctx);
     Preconditions.checkNotNull(request);
 
+    log.info("Executing streaming request '{}'", request);
```
Comment (xjdr) [inline src/main/java/com/google/spannerclient/GrpcClient.java]: Should probably be Debug?

File: src/main/java/com/google/spannerclient/Database.java (hunk 1 above)
Comment (pdex) [reply src/main/java/com/google/spannerclient/Database.java]: per chat: we're in alignment that we need to healthcheck the session objects in the pool.

File: src/main/java/com/google/spannerclient/GrpcClient.java (hunk 2 above)
Comment (pdex) [reply src/main/java/com/google/spannerclient/GrpcClient.java]: Yeah, I need to downgrade it or maybe move the log statement outside of the spannerclient.

**Pull request merged**